Make colors for challenge difficulty more easy to change

Rand0mLetters opened the pull request:
some colors got overwritten but now beginner difficulty is light blue, easy is green, medium is yellow, hard is light red, extreme is a very menacing red.

also made it easier to change the colors (now in styles/globals.css)

Hacksore commented:
@Rand0mLetters can you include some screenshots for us reviewers.

Rand0mLetters commented:
oh yeah for sure

I'm busy right now, I'll get them in the next 3-5h?

bautistaaa commented on apps/web/src/app/explore/_components/explore-card.tsx:
```diff
@@ -20,24 +20,24 @@ interface ExploreCardProps {
-    'dark:hover:border-yellow-300 hover:border-yellow-500 dark:group-focus:border-yellow-300 group-focus:border-yellow-500',
-  HARD: 'dark:hover:border-red-300 hover:border-red-500 dark:group-focus:border-red-300 group-focus:border-red-500',
+    'dark:hover:border-difficultyMediumDark hover:border-difficultyMedium dark:group-focus:border-difficultyMediumDark group-focus:border-difficultyMedium',
+  HARD: 'dark:hover:border-difficultyHardDark hover:border-difficultyHard dark:group-focus:border-difficultyHardDark group-focus:border-difficultyHard',
```
kebab case these

Rand0mLetters commented:
![image](https://github.com/typehero/typehero/assets/108293579/985b4625-c3f0-48a8-816a-603ab3e7c0f0)
![image](https://github.com/typehero/typehero/assets/108293579/b2a8c2b0-5fbc-467b-8608-daf36302504e)
![image](https://github.com/typehero/typehero/assets/108293579/55755956-8b3d-4d55-a8ac-5e7015ec66e9)
![image](https://github.com/typehero/typehero/assets/108293579/e125f267-6156-4bef-85fa-b2fd1a093906)
![image](https://github.com/typehero/typehero/assets/108293579/3a0e9741-7f51-4b63-9768-3275bc929666)
![image](https://github.com/typehero/typehero/assets/108293579/8fe9c303-87b6-4bc6-badd-0fc637bcc13c)
and in light mode (colors can be adjusted):
![image](https://github.com/typehero/typehero/assets/108293579/408e5dae-ee97-4574-a314-b66178448673)
![image](https://github.com/typehero/typehero/assets/108293579/39aeb537-fdc6-4082-9447-5dea896ea822)
![image](https://github.com/typehero/typehero/assets/108293579/b225d561-a918-4bb4-9d2a-9f2dea1952f0)
![image](https://github.com/typehero/typehero/assets/108293579/f69d8d64-37fa-4f22-b897-11c3dbcea678)
![image](https://github.com/typehero/typehero/assets/108293579/fdb403f6-fced-4529-a6ad-105e109eed61)

going to work on camelcasing the names in about 2 hours (my time estimation is really bad though, so it might be earlier)

PickleNik reviewed (changes requested):
Bad contrast ratios. Please use at least -600 for light mode and -400 or better -300 for dark

PickleNik commented:
Mainly light blue and light red in light mode are off.
And green and yellow don't look good on dark.

But I like the change of colors, although I'd rethink Hard and Extreme, maybe Orange (or keep it yellow as it was) for Mediums, Red as it was for Hards, and Purple/Indigo for Extremes?

Rand0mLetters commented:
I've updated the colors and converted them to hsl
They pass WCAG (except the light mode colors, on WCAG AAA with normal text, but it should(?) be fine, I think?)

The dark colors look very pastel-y, is that ok?

![image](https://github.com/typehero/typehero/assets/108293579/52c032db-846c-43ab-b31d-41f6a7a9ee5f)
![image](https://github.com/typehero/typehero/assets/108293579/273fbebf-76e7-48e3-beeb-6a0dba8cea43)
![image](https://github.com/typehero/typehero/assets/108293579/0cf7e6a2-2a71-4d3b-926a-4d86966ae83e)
![image](https://github.com/typehero/typehero/assets/108293579/c5b2d7ae-ab9b-471c-8d1c-96080a91148e)
![image](https://github.com/typehero/typehero/assets/108293579/6de25260-087a-45d5-8b19-b8c2a23d7e3d)


Dark mode (couldn't figure out the yellow color):
![image](https://github.com/typehero/typehero/assets/108293579/66245890-6930-4850-b3c5-07104791aac4)
![image](https://github.com/typehero/typehero/assets/108293579/d1d446be-3b5e-490f-88aa-6e688c2608c9)
![image](https://github.com/typehero/typehero/assets/108293579/eba20c1b-9674-4e08-949b-e182a38dc149)
![image](https://github.com/typehero/typehero/assets/108293579/9f1367b0-a745-44c0-8985-ecb64a7f960f)
![image](https://github.com/typehero/typehero/assets/108293579/f3296351-67d2-47d8-9542-79518e135006)

Rand0mLetters commented:
Oh shoot, have to update the colors on /tracks. Don't merge yet please

PickleNik commented:
this looks great, I would also make `HARD` and `MEDIUM` just 1 tone lighter on Dark mode, and `MEDIUM` one tone lighter on Light mode as well

PickleNik reviewed (approved):
do whatever u need to do for tracks but LGTM 👍🏼

Rand0mLetters commented:
just finished, running checks and pushing soon
spent too long figuring out how to use tailwind alpha values

Rand0mLetters commented:
![image](https://github.com/typehero/typehero/assets/108293579/6b4e3fef-57cc-4d32-8655-6918f3994555)
![image](https://github.com/typehero/typehero/assets/108293579/31a16991-de9e-4fc0-9f1f-1ff10e3eb9bb)

ready for merge (I think) @PickleNik
